tensorflow/core/api_def/base_api/api_def_ScatterNd.pbtxt

```diff
@@ -30,33 +30,28 @@ END
   }
   summary: "Scatters `updates` into a tensor of shape `shape` according to `indices`."
   description: <<END
-Update the input tensor by scattering sparse `updates` according to individual values at the specified `indices`.
-This op returns an `output` tensor with the `shape` you specify. This op is the
-inverse of the `tf.gather_nd` operator which extracts values or slices from a
-given tensor.
+Scatter sparse `updates` according to individual values at the specified
+`indices`. This op returns an output tensor with the `shape` you specify. This
+op is the inverse of the `tf.gather_nd` operator which extracts values or slices
+from a given tensor.
 
 This operation is similar to `tf.tensor_scatter_nd_add`, except that the tensor
-is zero-initialized. Calling `tf.scatter_nd(indices, values, shape)`
+is zero-initialized. Calling `tf.scatter_nd(indices, updates, shape)`
 is identical to calling
-`tf.tensor_scatter_nd_add(tf.zeros(shape, values.dtype), indices, values)`
+`tf.tensor_scatter_nd_add(tf.zeros(shape, updates.dtype), indices, updates)`
 
-If `indices` contains duplicates, the duplicate `values` are accumulated
-(summed).
+If `indices` contains duplicates, the associated `updates` are accumulated
+(summed) into the output tensor.
 
```

Comment:
The nondeterministic note is still accurate if determinism is not enabled. I would keep the note, but add a comment saying something like "However, the output will be deterministic if op determinism is enabled via `tf.config.experimental.enable_op_determinism`".

FYI, even in XLA the op is deterministic, since the Scatter HLO is deterministic when determinism is enabled (4dc6d4d59008b4558040afa1a5a5993f583bb48e). However, the SelectAndScatter HLO instruction, which is used by the MaxPoolGrad op, will raise an Unimplemented error when determinism is enabled.

Reply:
Thanks, Reed, for the reminder about those XLA-related changes and further clarification about how they affect other ops (i.e. MaxPoolGrad op determinism).

I have re-introduced a modified version of the note regarding nondeterminism.